### PR TITLE
Cache games without banners

### PR DESCRIPTION
**Pull request by JosJuice**
Games without banners were not cached before, because a banner could become available at any time, making the cache outdated without it becoming invalidated. Instead of not caching anything, this change makes Dolphin check for a banner every time a cache that lacks a banner is read. This is faster than reading all metadata, because reading a Wii banner only reads from the game's save file, not the volume and its filesystem.

The cache revision is incremented, because otherwise banners will be missing if a cache without a banner is created in the new version and the user switches to an old version and creates a savefile.

~~This change should also be applied to DolphinQt. Is anyone up for doing it? (cc @waddlesplash)~~

### CODE REVIEW

**Comment by waddlesplash** [comment]
The change doesn't look so complex, can you take a stab at implementing it on DQt? (I don't read #dolphin-dev logs very often, /msg me if you have any questions). If not, I might be able to have a look at it tomorrow. Maybe.

**Comment by JosJuice** [comment]
@waddlesplash The main problem I ran into is that DolphinQt sets `m_banner` to `Resources::GetPixmap(Resources::BANNER_MISSING)` when the banner is missing, unlike DolphinWX which makes `m_pImage` empty. If I just make the same changes to Qt as to wx, `BANNER_MISSING` will be stored in the cache as a valid non-empty banner, and then the constructor will think that the cache contains a real banner. It should be able to solve this by making `m_banner` be empty if there is no banner, like in wx, but I suppose that would require changes somewhere else in the code, and I don't know where.

**Comment by waddlesplash** [comment]
`m_pixmap` is private, IIRC, so you can just modify the getter function to return `BANNER_MISSING` if `m_pixmap` is null (or unset, or whatever), no?

**Comment by JosJuice** [comment]
Sounds good. I'll try to implement it.

File: Source/Core/DolphinQt/GameList/GameFile.h [hunk 1]
```diff
@@ -39,7 +41,12 @@ class GameFile final
 	u64 GetVolumeSize() const { return m_volume_size; }
 	// 0 is the first disc, 1 is the second disc
 	u8 GetDiscNumber() const { return m_disc_number; }
-	const QPixmap GetBitmap() const { return m_banner; }
+	const QPixmap GetBitmap() const {
```
Comment (lioncash) [inline Source/Core/DolphinQt/GameList/GameFile.h]: brace on next line.

File: Source/Core/DolphinQt/GameList/GameFile.h [hunk 2]
```diff
@@ -39,7 +41,12 @@ class GameFile final
...
+		if (m_banner.isNull())
+			return Resources::GetPixmap(Resources::BANNER_MISSING);
+		else
```
Comment (lioncash) [inline Source/Core/DolphinQt/GameList/GameFile.h]: This else is not necessary.

**Comment by waddlesplash** [comment]
Qt code LGTM.

**Comment by degasus** [comment]
LGTM

**Pull request merged**